addons/account/models/account_bank_statement.py

```diff
@@ -714,10 +457,111 @@ def _prepare_move_line_default_vals(self, counterpart_account_id=None):
         }
         return [liquidity_line_vals, counterpart_line_vals]
 
+    def _get_currencies(self):
+        """
+        Returns company, journal and foreign currencies of the lines
+        """
+        company_currency = self.journal_id.company_id.currency_id
+        journal_currency = self.journal_id.currency_id or company_currency
+        foreign_currency = self.foreign_currency_id or journal_currency or company_currency
+        return company_currency, foreign_currency, journal_currency
+
+    def _get_amounts(self):
+        """
+        Returns the line amount in company, journal and foreign currencies
+        """
+        company_currency, foreign_currency, journal_currency = self._get_currencies()
+        journal_amount = self.amount
+        if foreign_currency == journal_currency:
+            transaction_amount = journal_amount
+        else:
+            transaction_amount = self.amount_currency
+        if journal_currency == company_currency:
+            company_amount = journal_amount
+        elif foreign_currency == company_currency:
+            company_amount = transaction_amount
+        else:
+            company_amount = journal_currency._convert(journal_amount, company_currency, self.journal_id.company_id, self.date)
+        return company_amount, journal_amount, transaction_amount
+
     # -------------------------------------------------------------------------
     # COMPUTE METHODS
     # -------------------------------------------------------------------------
 
+    def _compute_running_balance(self):
+        line_map = {}
+
+        # Integrating the journal in the queries will make the query more complex, because we have different anchor
+        # points per journal, and it should be rare to have a read on multi journal in this model
+        self.flush_model()
+        for journal in self.journal_id:
+            journal_lines = self.filtered(lambda line: line.journal_id == journal)
+            anchor_line_domain = [
+                ('is_anchor', '=', True),
+                ('journal_id', '=', journal.id),
+            ]
+
+            min_index = min(journal_lines.mapped('internal_index'))
+            max_index = max(journal_lines.mapped('internal_index'))
+            if min_index :
+                anchor_line_domain.append(('internal_index', '<=', min_index))
+
+            anchor_index = self.search(
+                domain=anchor_line_domain,
+                order='internal_index desc',
+                limit=1
+            ).internal_index or ''
+            self._cr.execute(
+                f"""SELECT id, running_balance 
+                      FROM (
+                        SELECT st_line.id as id, statement_line_balance(amount, is_anchor, anchor_value) OVER (
```

Comment:
indent with 4 spaces plz

Reply:
It was too ugly with pure 4-spaces, so I used the guidelines in the upgrade repo, keeping the indents to 4 spaces where possible.